test.tf

```diff
@@ -108,24 +123,99 @@ resource "azurerm_postgresql_server" "example" {
   sku_name                     = "B_Gen5_2"
   storage_mb                   = 5120
   backup_retention_days        = 7
-  geo_redundant_backup_enabled = false
+  geo_redundant_backup_enabled = true
   auto_grow_enabled            = true
   administrator_login          = "DKFJnisdfu"
   administrator_login_password = "Aa12345678"
   version                      = "9.5"
   ssl_enforcement_enabled      = true
+
+  identity {
+    type = "SystemAssigned"
+  }
 }
 
 resource "azurerm_postgresql_configuration" "thrtottling_config" {
   name                = "connection_throttling"
   resource_group_name = "testrg"
   server_name         = azurerm_postgresql_server.example.name
-  value               = "off"
+  value               = "on"
 }
 
 resource "azurerm_postgresql_configuration" "example" {
   name                = "log_checkpoints"
   resource_group_name = "testrg"
   server_name         = azurerm_postgresql_server.example.name
-  value               = "off"
+  value               = "on"
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_postgresql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_postgresql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_postgresql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_postgresql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_mysql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_mysql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_mysql_active_directory_administrator" "example_adconfig" {
```

Comment:
**Accurics remediation**
 
**[MEDIUM]: Ensure geo-redundant backup is enabled for Azure PostgreSQL Server**
 
**Vulnerability**
Geo Redundant Backups are disabled on Azure PostgreSQL Server, which can have an impact on availability.
 
**Remediation**
Azure PostgreSQL Server provides the flexibility to choose between locally redundant or geo-redundant backup storage in the General Purpose and Memory Optimized tiers. When the backups are stored in geo-redundant backup storage, they are not only stored within the region in which your server is hosted, but are also replicated to a paired data center. This provides better protection and ability to restore your server in a different region in the event of a disaster. The Basic tier only offers locally redundant backup storage.

source: accurics-beta